intel_adsp: header cleanup and add soc specific headers

nashif opened the pull request:
Some cleanup but primarily introducing soc specific headers.


- intel_adsp: common: build soc.c only for CAVS
- intel_adsp: common: remove unnecessary ifndefs
- intel_adsp: move soc headers into include/
- intel_adsp: make shim header soc specific
- intel_adsp: shim: cleanup shim header
- intel_adsp: adsp_shim.h: run through clang-format
- intel-adsp: migreate cavs-mem.h to adsp_memory.h
- intel_adsp: move imr configs to headers

nashif committed:
intel_adsp: common: build soc.c only for CAVS

build soc.c only for platforms that can use it.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: common: remove unnecessary ifndefs

cavs15 has its own boot path, so no need to check for it in code it does
not run.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: move soc headers into include/

Move headers into the include/ folder per soc and rename fw_defs.h to
adsp_memory to align with SOF.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: make shim header soc specific

using once single header to support multiple socs and product
generations is error prone and not easily maintained.

Over time we have been adding conditional code in headers and extending
structs  to support new HW features which becomes a problem.

Goal is to keep platform headers in sync with hardware specification and
allow of introduction of new platforms and hardware features by just
introducing a new SoC with its own set of headers.

This is now just a copy of existing cavs-shim.h with slight changes,
goal is to clean this up long term and sync with hardware datasheets and
align on naming as well.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: shim: cleanup shim header

Remove conditional code and SOC specific defines.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: adsp_shim.h: run through clang-format

Run the files through clang-format to clean things up a bit.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel-adsp: migrate cavs-mem.h to adsp_memory.h

Move header and make it soc specific.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: move imr configs to headers

Move those defines and values back to headers. Kconfig is not a good
place for this, later this should move to DTS.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

edersondisouza commented on drivers/ipm/ipm_cavs_host.c:
```diff
@@ -3,7 +3,7 @@
  */
```
Commit message: s/migreate/migrate

nashif committed:
intel_adsp: move l1 memory definition to header

Move register definition to SoC header instead of hardcoding it where it
is being used.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: move register definition from soc.c to shim header

Declare those register in the adsp_shim header instead of the code
applying ifdefs that limit the scope to only specific SoCs.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: rename power_init_mtl

No need for the suffix here, this is built only for Meteorlake anyways.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: minor header include reorg/fixups

Minor reorg of headers and when they get included.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: unify timer registers and simplify timer driver

Declare clock control in the shim header per SoC and remove ifdeffry
from the driver simplifiying it and making it ready for the next
platform.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
intel_adsp: rename obsolete adsp timer config

Use INTEL_ADSP_TIMER instead.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
